```
startup_byt.c:560:3: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
  560 |   value global_data = caml_input_val(chan);
      |   ^~~~~
```
Since 5.0 intends to require a C11 compiler, it is a good time to get rid of this restriction @xavierleroy.

--- a/runtime/startup_byt.c
+++ b/runtime/startup_byt.c
@@ -556,9 +556,10 @@ CAMLexport void caml_main(char_os **argv)
   /* Load the globals */
   caml_seek_section(fd, &trail, "DATA");
   chan = caml_open_descriptor_in(fd);
-  /* TODO: do we need multicore Lock here */
-  caml_modify_generational_global_root(&caml_global_data, caml_input_val(chan));
-  /* TODO: do we need multicore Unlock here */
+  Lock(chan);
+  value global_data = caml_input_val(chan);
+  Unlock(chan);
+  caml_modify_generational_global_root(&caml_global_data, global_data);
   caml_close_channel(chan); /* this also closes fd */
   caml_stat_free(trail.section);
   /* Initialize system libraries */

